src/sphinx/reference/adapters.rst

```diff
@@ -54,18 +54,221 @@ For persisting the stream processing progress, an application should store the l
+
+Applications invoke the ``start`` method of the system to initialize the registered event producers and start event exchange with the ``Vert.x`` instance.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: adapter-example
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: adapter-example
+
+.. warning::
+   The ``start`` method should only be called once all handlers on the event bus have been registered. Failing to do so may lead to loss of events because an producer might try to deliver events to an event bus endpoint which has not yet an event handler assigned to it.
+
+The following sections contain a detailed description of the different kinds of event producers.
+
+Vert.x Publish Event Producer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A *Publish Event Producer* publishes events from an event log to *multiple* subscribers on the event bus. Events are delivered to specific endpoints defined in the configuration of the producer. A producer can route events to different event bus endpoints based on the content of the event. Event routing is enabled by supplying a partial function which maps events to event bus endpoints. If no endpoint is returned by the function the event will not be processed.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: vertx-publish-producer
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: vertx-publish-producer
+
+Event delivery is always performed with *At-Most-Once* delivery semantics, so no guarantees about the successful delivery of events can be made.
+
+Applications consume events by registering an event bus handler at the configured endpoints.
+
+.. tabbed-code::
+   .. includecode:: ../../../eventuate-example-vertx/src/main/scala/com/rbmhtechnology/docs/vertx/Documentation.scala
+      :snippet: event-processing-vertx-producer
+   .. includecode:: ../../../eventuate-example-vertx/src/main/java/com/rbmhtechnology/docs/vertx/japi/Documentation.java
+      :snippet: event-processing-vertx-producer
+
+Replication progress from the source event log is tracked by persisting the ``localSequenceNr`` of the latest delivered event to the ``StorageProvider`` supplied to the ``VertxAdapter``. After delivery of one or multiple events the replication progress is persisted. The producer continues delivery of events from the latest known ``localSequenceNr`` once the it is started.
```

Comment:
> Replication progress

read progress

> latest delivered event

The latest *sent* event

Delivery usually means delivery to the consumer but in this context you can only reason about successful sending/publishing to the event bus. So other occurrences of *deliver...* should be replaced as well here.